tests/src/Context/ModuleContext.php

```diff
@@ -0,0 +1,149 @@
+<?php
+/**
+ * @file
+ * Contains \Drupal\nexteuropa\Context\ModuleContext.
+ */
+
+namespace Drupal\nexteuropa\Context;
+
+use Drupal\DrupalExtension\Context\RawDrupalContext;
+use Behat\Gherkin\Node\TableNode;
+use function bovigo\assert\assert;
+use function bovigo\assert\predicate\isEmpty;
+
+/**
+ * Context with module, feature and feature_set management.
+ */
+class ModuleContext extends RawDrupalContext {
+
+  /**
+   * Initial module list to restore at the end of the test.
+   *
+   * @var array
+   */
+  protected $initialModuleList = array();
+
+
+  /**
+   * Stores the list of enabled modules before executing a scenario.
+   *
+   * @BeforeScenario
+   */
+  public function storeDefaultEnabledModules() {
+    $this->initialModuleList = module_list(TRUE);
+  }
+
+  /**
+   * Restores the initial values of the Drupal Modules.
+   *
+   * @AfterScenario
+   */
+  public function restoreInitialState() {
+    $after_scenario_modules = module_list(TRUE);
+
+    $lists_diff = array_diff($after_scenario_modules, $this->initialModuleList);
+
+    if ($lists_diff) {
+      module_disable($lists_diff);
+      drupal_uninstall_modules($lists_diff);
+      drupal_flush_all_caches();
+
+      // Check if modules are really uninstalled.
+      module_list(TRUE);
+      foreach ($lists_diff as $module) {
+        if (module_exists($module)) {
+          throw new \Exception(sprintf('Module "%s" could not be uninstalled', implode(', ', $module)));
```

Comment:
Like as I said in other PR, here it is a preparation step and as nothing to do with the test itself.

It is not related with the scenario itself. We come back in simple coding and when we meet exceptional event, we trigger an exception.

In the other case, we are in the scenario test. We really test an event in relation with the scenario. Then, semantically, the use of an "assert" is justified.